pkg/router/template/template_helper.go

```diff
@@ -347,9 +351,20 @@ func clipHAProxyTimeoutValue(val string) string {
 			return haproxyMaxTimeout
 		}
 	} else {
-		duration, err := time.ParseDuration(val)
+		var duration, err = templateutil.ParseDurationWithErrors(val)
 		if err != nil {
-			return val
+			// Here we need to check if the value is greater than what time.ParseDuration
+			// allows (value must be representable as uint64, e.g. approximately 2562047.79 hours
+			// or 9223372036854775807 nanoseconds).
+			if strings.Contains(err.Error(), "overflow") {
+				log.V(7).Info("Route annotation timeout exceeds maximum allowable by HAProxy, clipping to max")
+				return haproxyMaxTimeout
+			} else if strings.Contains(err.Error(), "missing unit") {
+				// ignore this error, because we have allowed this in the past
```

Comment:
What if `val` is "9999999999999999"?

Reply:
i will add the missing "ms" that HAProxy tacks on when the unit is missing instead.

Reply:
I was concerned that ignoring the error and passing "9999999999999999" through would result in an unparseable `haproxy.config`.  Looks like you handle this case properly now and cap "9999999999999999" to the maximum value now.